conda/core/index.py

```diff
@@ -52,104 +57,186 @@ def get_index(channel_urls=(), prepend=True, platform=None,
     if prefix:
-        _supplement_index_with_prefix(index, prefix, known_channels)
+        _supplement_index_with_prefix(index, prefix)
     if unknown:
-        _supplement_index_with_cache(index, known_channels)
+        _supplement_index_with_cache(index)
     if context.track_features:
         _supplement_index_with_features(index)
     return index
 
 
 def fetch_index(channel_urls, use_cache=False, index=None):
-    # type: (prioritize_channels(), bool, bool, Dict[Dist, IndexRecord]) -> Dict[Dist, IndexRecord]
     log.debug('channel_urls=' + repr(channel_urls))
 
     use_cache = use_cache or context.use_index_cache
 
     # channel_urls reversed to build up index in correct order
-    CollectTask = namedtuple('CollectTask', ('url', 'schannel', 'priority'))
-    tasks = (CollectTask(url, *channel_urls[url]) for url in reversed(channel_urls))
-    index = collect_all_repodata_as_index(use_cache, tasks)
+    from .repodata import collect_all_repodata_as_index
+    index = collect_all_repodata_as_index(use_cache, channel_urls)
 
     return index
 
 
-def _supplement_index_with_prefix(index, prefix, channels):
-    # type: (Dict[Dist, IndexRecord], str, Set[canonical_channel]) -> None
+def _supplement_index_with_prefix(index, prefix):
     # supplement index with information from prefix/conda-meta
     assert prefix
-    maxp = len(channels) + 1
-    for dist, info in iteritems(linked_data(prefix)):
+    for dist, prefix_record in iteritems(linked_data(prefix)):
         if dist in index:
             # The downloaded repodata takes priority, so we do not overwrite.
             # We do, however, copy the link information so that the solver (i.e. resolve)
             # knows this package is installed.
-            old_record = index[dist]
-            link = info.get('link') or EMPTY_LINK
-            index[dist] = IndexRecord.from_objects(old_record, link=link)
+            current_record = index[dist]
+            link = prefix_record.get('link') or EMPTY_LINK
+            index[dist] = PrefixRecord.from_objects(current_record, prefix_record, link=link)
         else:
-            # If the package is not in the repodata, use the local data. If
-            # the 'depends' field is not present, we need to set it; older
-            # installations are likely to have this.
-            depends = info.get('depends') or ()
+            # If the package is not in the repodata, use the local data.
             # If the channel is known but the package is not in the index, it
             # is because 1) the channel is unavailable offline, or 2) it no
             # longer contains this package. Either way, we should prefer any
             # other version of the package to this one. On the other hand, if
             # it is in a channel we don't know about, assign it a value just
             # above the priority of all known channels.
-            priority = MAX_CHANNEL_PRIORITY if dist.channel in channels else maxp
```

Comment:
I know the channel priority assignments have been moved, but I want to know in particular how we ensure under the new regime that these orphan packages get marked as deprecated. Assigning MAX_CHANNEL_PRIORITY here was doing that, but since this step is skipped, I'm worried that somewhere downstream we're just seeing its channel and thinking it is still part of the index.